### PR TITLE
Vipinhas/capacity res tiers

### PR DESCRIPTION
**Pull request by vipinhas**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Please ensure to add changelog with this PR by answering the following questions.
  1. What's the purpose of the update?    

      - [x] new API version 

  2. When you are targeting to deploy new service/feature to public regions? Please provide date, or month to public if date is not available yet.  June 2, 2021
  
  3. When you expect to publish swagger? Please provide date, or month to public if date is not available yet. June 2, 2021
  4. If it's an update to existing version,  please select SDKs of specific language and CLIs that require refresh after swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [x] PowerShell
      - [x] CLI
      - [ ] Terraform
      - [ ] No, no need to refresh for updates in this PR

### Contribution checklist:
- [ ] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [ ] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [x] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  -  [x] Ensure to copy the existing version into new directory structure for first commit (including refactoring) and then push new changes including version updates in separate commits. This is required to review the changes efficiently.
  - Adding a new service

- [ ] Please ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by vipinhas** [comment]
Hi  @akning-ms,

I see that some validations are failing, can you pls help me understand the error?
on Avocado - it complain on INCONSISTENT_API_VERSION although I didnt detect any inconcinnities  
Regarding MaxCapacity, it a read only property that was removed.

I need to advance the PR asap, I'd appreciate your help with it

File: specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2020-10-01/Workspaces.json [hunk 1]
```diff
@@ -424,12 +424,6 @@
           "format": "int32",
           "description": "The capacity reservation level for this workspace, when CapacityReservation sku is selected."
         },
-        "maxCapacityReservationLevel": {
-          "type": "integer",
-          "format": "int32",
-          "description": "The maximum capacity reservation level available for this workspace, when CapacityReservation sku is selected.",
-          "readOnly": true
-        },
         "lastSkuUpdate": {
```
Comment (akning-ms) [inline specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2020-10-01/Workspaces.json:427]: Can I know what is the reason to remove it? it is a breaking change which need to breaking change review board to approval

Comment (akning-ms) [reply specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2020-10-01/Workspaces.json:427]: > Hi @akning-ms,
> 
> I see that some validations are failing, can you pls help me understand the error?
> on Avocado - it complain on INCONSISTENT_API_VERSION although I didnt detect any inconcinnities
> Regarding MaxCapacity, it a read only property that was removed.
> 
> I need to advance the PR asap, I'd appreciate your help with it

It is due to the version in info section is wrong as below:
  "info": {
    "title": "Azure Log Analytics",
    "description": "Azure Log Analytics API reference",
    "version": "**2020-10-01**"
  },

Comment (vipinhas) [reply specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2020-10-01/Workspaces.json]: 1. Sure, The reason for the readonly  maxCapacityRemoval is a change in pricing tier for log analytics workspace & cluster.  (the propery was already removed for ws ->https://github.com/Azure/azure-rest-api-specs/pull/13812 ).
Today with the new api-version we are introducing new enum for  simplification of the billing process and avoid over pricing on overage.

2. Thanks, fixed

Comment (vipinhas) [reply specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2020-10-01/Workspaces.json]: Breaking change template (Approved) -> https://msazure.visualstudio.com/One/_workitems/edit/10050734

File: specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2020-10-01/Clusters.json [hunk 2]
```diff
@@ -629,12 +629,6 @@
           "description": "Minimum CapacityReservation value in GB.",
           "type": "integer",
           "format": "int64"
-        },
-        "maxCapacity": {
-          "readOnly": true,
-          "description": "Maximum CapacityReservation value in GB.",
-          "type": "integer",
-          "format": "int64"
         }
```
Comment (akning-ms) [inline specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2020-10-01/Clusters.json:632]: This removing caused model validation failure, you may also change example files.
and can you also send me the PR# which get approval for breaking change?

Comment (vipinhas) [reply specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2020-10-01/Clusters.json:632]: Thanks, fixed. 
Link to PR LA workspace breaking change-> https://github.com/Azure/azure-rest-api-specs/pull/13812

File: specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2020-10-01/Workspaces.json (hunk 1 above)
Comment (akning-ms) [reply specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2020-10-01/Workspaces.json]: Looks the approval is not about "maxCapacity"

File: specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2020-10-01/Clusters.json (hunk 2 above)
Comment (akning-ms) [reply specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2020-10-01/Clusters.json:632]: Looks the approval is not about "maxCapacity"

Comment (vipinhas) [reply specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2020-10-01/Clusters.json:632]: The property is named a bit different between workspace & cluster resources but operates the same -> limit capacity reservation values. We have introduced new pricing tiers to remove this limit, reduce costs and offer selection for the customer (Enum)

Comment (vipinhas) [reply specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2020-10-01/Clusters.json:632]: @akning-ms , any update? can you advise on how to move on?

**Comment by vipinhas** [comment]
Hey Arthur,

There was no action on the ticket for while, can you advise on how to proceed?
link to PR - Vipinhas/capacity res tiers by vipinhas · Pull Request #14598 · Azure/azure-rest-api-specs (github.com)<https://github.com/Azure/azure-rest-api-specs/pull/14598>

Thanks,
Victoria

________________________________

**Comment by akning-ms** [comment]
Hi @vipinhas, this PR is currently pending on ARM review. you can refer to first comment for below guide:
**If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them.**

**Comment by chiragg4u** [comment]
Please ensure you've reviewed following guidelines including ARM resource provider contract and REST guidelines. Estimated time (4 hours). This is required before you can request review from ARM API Review board.

**Comment by chiragg4u** [comment]
Looks like you are not following the below guideline. Without this review can not continue. Please cancel this PR and create new one following the guidance.
Ensure to copy the existing version into new directory structure for first commit (including refactoring) and then push new changes including version updates in separate commits. This is required to review the changes efficiently.

**Pull request closed without merging**